source/common/stats/raw_stat_data.h

```diff
@@ -95,17 +95,56 @@ struct RawStatData {
 
 using RawStatDataSet = BlockMemoryHashSet<Stats::RawStatData>;
 
+template <class Stat> class RawStat : public Stat {
+public:
+  RawStat(StatName stat_name, RawStatData& data, StatDataAllocatorImpl<RawStatData>& alloc,
+          absl::string_view tag_extracted_name, const std::vector<Tag>& tags)
+      : Stat(data, alloc, tag_extracted_name, tags),
+        stat_name_storage_(stat_name, alloc.symbolTable()) {}
+  ~RawStat() { stat_name_storage_.free(this->symbolTable()); }
+
+  StatName statName() const override { return stat_name_storage_.statName(); }
+
+private:
+  StatNameStorage stat_name_storage_;
+};
+
 class RawStatDataAllocator : public StatDataAllocatorImpl<RawStatData> {
 public:
   RawStatDataAllocator(Thread::BasicLockable& mutex, RawStatDataSet& stats_set,
                        const StatsOptions& options, SymbolTable& symbol_table)
       : StatDataAllocatorImpl(symbol_table), mutex_(mutex), stats_set_(stats_set),
         options_(options) {}
+  ~RawStatDataAllocator();
+
+  virtual RawStatData* alloc(absl::string_view name); // Virtual only for mocking.
+  void free(Stats::RawStatData& data) override;
+  RawStatData* allocStatName(StatName stat_name) {
+    return alloc(symbolTable().toString(stat_name));
+  }
 
   // StatDataAllocator
   bool requiresBoundedStatNameSize() const override { return true; }
-  Stats::RawStatData* alloc(absl::string_view name) override;
-  void free(Stats::RawStatData& data) override;
+
+  template <class Stat>
+  std::shared_ptr<Stat> makeStat(StatName name, absl::string_view tag_extracted_name,
```

Comment:
Documentation on this new makeStat() please, since it seems to be a hero function in threading this through elsewhere.

Reply:
I think @fredlas #5910 will merge first, which is likely to delete this file so I'l hold off tweaking it.